Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/EOAWalletToSequenceWalletAdapter.cs

```diff
@@ -0,0 +1,275 @@
+        }
+
+        public event Action<string> OnSignMessageComplete;
+        public event Action<string> OnSignMessageFailed;
+
+        public async Task<string> SignMessage(Chain network, string message, uint timeBeforeExpiry = 30)
+        {
+            byte[] byteChain = Encoding.UTF8.GetBytes(network.GetChainId());
+            byte[] byteMessage = Encoding.UTF8.GetBytes(message);
+
+            try
+            {
+                var signature = await _wallet.SignMessage(byteMessage, byteChain);
+                OnSignMessageComplete?.Invoke(signature);
+                return signature;
+
+            }
+            catch (Exception e)
+            {
+                OnSignMessageFailed?.Invoke("Error signing message: "+ e.Message);
+                return e.Message;
+            }
+        }
+
+        public async Task<IsValidMessageSignatureReturn> IsValidMessageSignature(Chain network, string message, string signature)
+        {
+            try
+            {
+                return new IsValidMessageSignatureReturn( await _wallet.IsValidSignature(signature, message, network));
+            }
+            catch (Exception e)
+            {
+               throw  new Exception("An error occurred while validating the message signature.", e);
+            }
+        }
+
+        public event Action<SuccessfulTransactionReturn> OnSendTransactionComplete;
+        public event Action<FailedTransactionReturn> OnSendTransactionFailed;
+
+        public async Task<TransactionReturn> SendTransaction(Chain network, Transaction[] transactions, bool waitForReceipt = true, uint timeBeforeExpiry = 30)
+        {
+            var client = new SequenceEthClient(network);
+            EthTransaction[] ethTransactions = new EthTransaction[transactions.Length];
+            List<FailedTransactionReturn> failedTransactionReturns = new List<FailedTransactionReturn>();
+            List<SuccessfulTransactionReturn> successfulTransactionReturns = new List<SuccessfulTransactionReturn>();
+
+            for (int i = 0; i < transactions.Length; i++)
+            {
+                EthTransaction transaction;
+                string toAddress;
+                BigInteger value;
+                string data = null;
+                switch (transactions[i])
+                {
+                    case RawTransaction tx:
+                        toAddress = tx.to;
+                        value = BigInteger.Parse(tx.value);
+                        data = tx.data;
+                        transaction = await new GasLimitEstimator(client, _wallet.GetAddress())
+                                                                   .BuildTransaction(toAddress, data, value);
+                        break;
+
+                    case SendERC20 tx:
+                        toAddress = tx.to;
+                        value = BigInteger.Parse(tx.value);
+                        transaction = await new GasLimitEstimator(client, _wallet.GetAddress())
```

Comment:
This should be an erc20 transaction - you should be creating an ERC20 object as you've done with the other tokens